core/src/services/dbfs/reader.rs

```diff
@@ -0,0 +1,169 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::cmp;
+use std::io::SeekFrom;
+use std::sync::Arc;
+use std::task::ready;
+use std::task::Context;
+use std::task::Poll;
+
+use async_trait::async_trait;
+use base64::engine::general_purpose;
+use base64::Engine;
+use bytes::BufMut;
+use bytes::Bytes;
+use futures::future::BoxFuture;
+use serde::Deserialize;
+
+use super::core::DbfsCore;
+
+use crate::raw::*;
+use crate::*;
+
+const DBFS_READ_LIMIT: usize = 1048576;
```

Comment:
Please add a reference for this limist.

Reply:
Add note and reference, and changed it to `1024 * 1024`.